addons/mail/models/discuss/discuss_channel.py

```diff
@@ -250,6 +252,39 @@ def _compute_self_member_id(self):
         for channel in self:
             channel.self_member_id = member_by_channel.get(channel)
 
+    @api.depends_context("uid", "guest")
+    @api.depends("self_member_id", "channel_member_ids", "channel_member_ids.channel_role")
```

Comment:
It doesn't depend on `channel_member_ids`, just on `self_member_id.channel_role`